test/TestProjects/Models-Cadl/Generated/FixedIntEnum.Serialization.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+
+namespace ModelsInCadl
+{
+    internal static partial class FixedIntEnumExtensions
```

Comment:
One thing we could optimize is to delete the serialization class for fixed int enum, since it's int value by nature in .net. However, doing so will result in changes across various places in the generator. Some changes are non-trivial.

Reply:
We've decided to not delete the serialization class for fixed int enum. However, we will not need serialization method here, and replace it with type cast, e.g. `WriteNumber((int)some_enum)`;